pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TransactionBufferHandlerImpl.java

```diff
@@ -336,8 +291,42 @@ protected OpRequestSend newObject(Handle<OpRequestSend> handle) {
         };
     }
 
+    public CompletableFuture<ClientCnx> getClientCnxWithLookup(String topic) {
+        return pulsarClient.getConnection(topic);
+    }
+
     public CompletableFuture<ClientCnx> getClientCnx(String topic) {
-        return ((PulsarClientImpl) pulsarClient).getConnection(topic);
+        NamespaceService namespaceService = pulsarService.getNamespaceService();
+        CompletableFuture<NamespaceBundle> nsBundle = namespaceService.getBundleAsync(TopicName.get(topic));
+        return nsBundle
+                .thenCompose(bundle -> namespaceService.getOwnerAsync(bundle))
```

Comment:
how about use getBrokerServiceUrlAsync?

Reply:
Yes, right, we can avoid using `getOwnerAsync`.

Reply:
Rollback to this implementation, because `getBrokerServiceUrlAsync ` will use LookupOptions which has an `authoritative` param that will cause some tests failed.